src/mercury.c

```diff
@@ -655,11 +655,6 @@ hg_set_struct(struct hg_private_handle *hg_handle,
             ret = HG_INVALID_PARAM;
             goto done;
     }
-    if (!proc_cb || !struct_ptr) {
-        /* Silently skip */
-        *payload_size = 0;
```

Comment:
Instead of the changes that you've made, could you please try to change that line with `*payload_size = header_offset`

Reply:
Sure thing.  I tested that suggestion and it works fine; PR has been updated accordingly.

Reply:
Great thanks!